### ISSUE TITLE
Install python binding locally / Pip package

### ISSUE THREAD

**Issue by jdumas** [libigl/libigl#553]
Hi,

I was wondering if it would be interesting to provide a simple way to install libigl's python bindings to the system (installing them locally in userspace). That would remove the need for the awkards `sys.path.insert(0, os.getcwd() + "/../")` that we see in the tutorials, and allow one to use `import pyigl` out of the box.

A clean way to provide information about where to find `pyigl` would be to use path configuration files, as explained in [this SO answer](https://stackoverflow.com/a/12311321/1419074):

```bash
# find directory
SITEDIR=$(python -m site --user-site)

# create if it doesn't exist
mkdir -p "$SITEDIR"

# create new .pth file with our path
echo "$HOME/foo/bar" > "$SITEDIR/somelib.pth"
```

I imagine this could easily be adapted for libigl (and as a python script to work on Windows as well). E.g. one could have a script `python/install.py` that would do what's above.

**Comment by danielepanozzo** [comment]
I don't know what is the most common way to handle python libraries. Wouldn't it be better to add them to pip so that they can be downloaded and installed directly (including in python virtual environments)?

**Comment by jdumas** [comment]
That would be ideal, but the standard way to do it (via `setuptools`) would require more substantial changes to the current build process (I don't have experience with `setuptools` so I don't know exactly what needs to be done). I'm proposing this as a quick way to install the library locally given the way it's currently built.

**Comment by danielepanozzo** [comment]
Mmm I would prefer the long-term solution if possible. Let's wait for @skoch9 to comment on this. We could also ask @panchagil , she has lots of experience with Python and maybe she can take care of the "setuptools" script.

**Comment by skoch9** [comment]
I would also vote for pip for the installation. I have done this already for a couple projects (only for pure python packages though).  I think if we go for a binary distribution (with wheels) we don't have to change much to the current build process. We could use the pip development mode for local development installations and update the wheel distribution package from time to time. We just have to figure out a few things first (how to handle the nanogui dependency, what should be included in the build, etc.).
I could look into this from next week on...

**Comment by panchagil** [comment]
Hi,

I have a tiny project that uses setuptools to install a dummy pybind c++ module. ([here](https://github.com/panchagil/boostraps/tree/master/pybind_eigen))

My code is based on the _building with setuptools_ examples mentioned in the pybind11 documentation ([here](http://pybind11.readthedocs.io/en/stable/compiling.html?highlight=setuptools)). I added a couple of lines to get the correct python, since cmake doesn't guess very well when using virtualenvs. I've tested it only with python 3 and only on develop mode using `python setup.py develop`. 

There might be some path issues when calling it as `pip install pymodule`. But it might offer a good starting point if anyone wants to work on it. I wanted to improve on it to use it to install libigl and to install c++ code that uses libigl, but I haven't had time to do it.
 
The CMake is based on the one on libigl. I have less experience with that part :)
